### DISCUSSION TITLE
prevent spacebar miss from interrupting flow

### DISCUSSION THREAD

**Discussion by givensuman** [monkeytypegame/monkeytype#1318] [Ideas]
I think it would be nice if missing a space hit didn't ruin the flow of typing. If you miss a key normally it "ruins" the word for you in that if you keep on typing it doesn't count as a correct word. Normally this is fine as one can compensate a lost word for the added speed of not going back to fix it. If you miss a space, however, all subsequent words are ruined until backspacing to the spot in which you missed it.

I think there should be a smart recognition system in which if a user misses a space but correctly types the next word in the list, the mistake is forgiven and the flow of typing is uninterrupted. The word missing a space after it would still be flagged as wrong.

**Comment by frogfrogfrogfrogfrog** [comment]
From my understanding this could be seen as some sort of "cheat" where you don't have to hit the space bar (as long as you type the words correctly). I figure that a solution to this would be "nospace" or perhaps "stop on word/letter". Also if it's your accuracy that's holding you back then perhaps your accuracy could be improved?

**Reply by givensuman** [reply]
To clarify, this is to treat a spacebar hit like other keys, simply making the games own rules more consistent.

For example, if you have:
monkey type is cool 
And type (/ representing a spacebar hit):
monkye/type/is/cool
You missed the ordering of a couple letters in a word and it only counts against you for that word. With a spacebar, hitting it anywhere incorrectly counts against you for subsequent words, i.e.:
monke/ytype/is/cool

When you are typing 180+ wpm mistakes are inevitable, stopping your flow to hit backspace is impossible afaik

**Reply by frogfrogfrogfrogfrog** [reply]
Ah that makes more sense. Although I still don't see how this could fix anything. As @Miodec said, "Both have the possibility of misalignment between what youre typing and whet youre supposed to type".

**Comment by Miodec** [comment]
Treating the spacebar as a character instead of a "word break" will cause more confusion than currently in my opinion.

Both have the possibility of misalignment between what youre typing and what youre supposed to type and this change is not worth the effort.
